internal/configs/configmaps.go

```diff
@@ -501,6 +509,33 @@ func ParseConfigMap(cfgm *v1.ConfigMap, nginxPlus bool, hasAppProtect bool) *Con
 		}
 	}
 
+	if hasAppProtectDos {
+
+		appProtectDosLivenessEnable, livenessExistsEnable, errEnable := GetMapKeyAsBool(cfgm.Data, "app-protect-dos-liveness-enable", cfgm)
+		appProtectDosLivenessURI, livenessExistsURI := cfgm.Data["app-protect-dos-liveness-uri"]
+		appProtectDosLivenessPort, livenessExistsPort, errPort := GetMapKeyAsInt(cfgm.Data, "app-protect-dos-liveness-port", cfgm)
+
+		if livenessExistsEnable {
+			if errEnable != nil {
```

Comment:
I'm not 100% about this logic, I think it's worth checking it works. I normally see the following pattern in Golang:
```
value, err := getValue()
if err != nil {
   ... handle error
}
...use value
```

What happens in this case if there's an `errEnable`, will livenessExistsEnable ever be true in which case the error won't be logged and found ?

Reply:
Change logic, handle first err and then the usecase